python/pyiceberg/avro/resolver.py

```diff
@@ -97,6 +119,82 @@ def construct_reader(
     return resolve(file_schema, file_schema, read_types)
 
 
+def construct_writer(file_schema: Union[Schema, IcebergType]) -> Writer:
+    """Constructs a writer from a file schema.
+
+    Args:
+        file_schema (Schema | IcebergType): The schema of the Avro file.
+
+    Raises:
+        NotImplementedError: If attempting to resolve an unrecognized object type.
+    """
+    return visit(file_schema, ConstructWriter())
+
+
+class ConstructWriter(SchemaVisitorPerPrimitiveType[Writer]):
+    """Constructs a writer tree from an Iceberg schema."""
+
+    def schema(self, schema: Schema, struct_result: Writer) -> Writer:
+        return struct_result
+
+    def struct(self, struct: StructType, field_results: List[Writer]) -> Writer:
+        return StructWriter(tuple(field_results))
+
+    def field(self, field: NestedField, field_result: Writer) -> Writer:
+        if field.required:
```

Comment:
Just a suggestion, I like to inline this kind of statement:
```suggestion
        return field_result if field.required else OptionWriter(field_result)
```
Feel free to ignore this